internal/controller/prometheus_manager.go

```diff
@@ -61,28 +58,21 @@ func RegisterMetrics() {
 }
 
 // UpdateGpuSliceMetrics updates GPU slice allocation metrics
-func (r *InstasliceReconciler) IncrementTotalProcessedGpuSliceMetrics(instasliceObj inferencev1alpha1.Instaslice, nodeName, gpuID string, profile string, pod *v1.Pod) error {
-	processedSlices, err := r.calculateProfileFitOnGPU(&instasliceObj, profile, gpuID, false, pod)
-	if err != nil {
-		return fmt.Errorf("failed to calculate processed GPU slices for profile %s: %w", profile, err)
-	}
-	instasliceMetrics.processedSlices.WithLabelValues(nodeName, gpuID).Add(float64(processedSlices))
-	return nil
+func (r *InstasliceReconciler) IncrementTotalProcessedGpuSliceMetrics(nodeName, gpuID string, profile string, size int32) {
+	instasliceMetrics.processedSlices.WithLabelValues(nodeName, gpuID).Add(float64(size))
 }
 
 // UpdateGpuSliceMetrics updates GPU slice allocation metrics
-func (r *InstasliceReconciler) UpdateDeployedPodTotalMetrics(nodeName, gpuID, namespace, podname, profile string, size int32) error {
```

Comment:
What is the reason to not return the error?
IMO, you can still return the error from here and add a log in the actual Reconcile loop when error occurs

Reply:
Prometheus doesn't return errors.

Reply:
Right, prometheus may not return error, but there could be new errors created based on the situation say [this](https://github.com/openshift/instaslice-operator/pull/490/files#diff-b5b4748e8599ec08003b932bf442dbe8f9835a73cd35719ac96fa5bb269e93c7L76)
I think it would be a good practice to return errors and log so that we don't miss a situation there.
Here in this case without an error, we don't know if the metric is updated or just returned silently.